File: LeetCode/Simulation/Python/Leetcode2367-number-of-arithmetic-triplets.py
class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        st, ans = set(), 0
        for x in nums: st.add(x)
        for x in nums:
            if x + diff in st and x + 2 * diff in st:
                ans += 1
        return ans

File: LeetCode/Simulation/Rust/Leetcode2367-number-of-arithmetic-triplets.rs
impl Solution {
  pub fn arithmetic_triplets(nums: Vec<i32>, diff: i32) -> i32 {
    use std::collections::HashSet;
    let mut st = HashSet::new();
    for x in nums.iter() {
      st.insert(x);
    }
    let mut ans = 0;
    for x in nums.iter() {
      if st.contains(&(x + diff)) && st.contains(&(x + 2 * diff)) {
        ans += 1;
      }
    }
    ans
  }
}
